### DISCUSSION TITLE
RedissonLock.tryLock is throwing StacklessClosedChannelException

### DISCUSSION THREAD

**Discussion by shaharbelizon** [redisson/redisson#5985] [General]
Hi,
We are using Redisson v3.31.0 within a spring-boot application (v3.2.5) with Java 17, and we are facing the following error from time to time:
java.util.concurrent.ExecutionException: io.netty.channel.StacklessClosedChannelException
at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:396)
at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:2096)
at org.redisson.RedissonLock.tryLock(RedissonLock.java:250)

Any idea why it happened? How can we avoid this kind of exception?

**Comment by mrniko** [comment]
Did you check the network, Redis and application CPU consumption? 
It might be connected to https://github.com/redisson/redisson/issues/5553 can you reproduce it with 3.32.0 ?

**Reply by shaharbelizon** [reply]
Thanks for the quick reply!
Our Redis server is configured with 3 primary nodes and 3 replicas, ~9000 commands/s each node, Net In ~1700 kb/s each node, Net Out ~1200 kb/s each node, total Memory ~60Mb each node,  ~600 connections each node, and a total of 10K keys.
The Redis CPU cluster consumption is ~130% in total (I guess for 6 nodes it's ok).
As for version 3.32.0, I didn't see such a version in the maven repo, there is only 3.31.0, and since we are talking about a production environment, I preferred to include only released versions.
Thanks, Shahar

**Reply by mrniko** [reply]
Here it is https://repo1.maven.org/maven2/org/redisson/redisson/3.32.0/

Is there any throughput limit applied by Redis vendor? It also causes connections to be closed.

**Reply by shaharbelizon** [reply]
Thanks, @mrniko, I just saw the new version released yesterday, I will try it :)
